website/www/site/content/en/contribute/_index.md

```diff
@@ -57,46 +57,98 @@ There are lots of opportunities to contribute. You can for example:
-   changes locally.
- - For SDK Development:
-      - [Go](https://golang.org) 1.12 or later installed for Go SDK development
-      - Python 3.x interpreters. You will need Python interpreters for all Python versions supported by Beam.
-        Interpreters should be installed and available in shell via `python3.x` commands. For more information, see:
-        Python installation tips in [Developer Wiki](https://cwiki.apache.org/confluence/display/BEAM/Python+Tips#PythonTips-InstallingPythoninterpreters).
- - For large contributions, a signed [Individual Contributor License
-   Agreement](https://www.apache.org/licenses/icla.pdf) (ICLA) to the Apache
-   Software Foundation (ASF).
-
-
-### Configuration options
+- A [GitHub](https://github.com/) account.
+- A Linux, macOS, or Microsoft Windows development environment
+- Java JDK 8 installed
+- [Docker](https://www.docker.com/) installed for some tasks including building worker containers and testing this website
+  changes locally.
+- For SDK Development:
+   - [Go](https://golang.org) 1.12 or later installed for Go SDK development
+   - Python 3.x interpreters. You will need Python interpreters for all Python versions supported by Beam.
+     Interpreters should be installed and available in shell via `python3.x` commands. For more information, see:
+     Python installation tips in [Developer Wiki](https://cwiki.apache.org/confluence/display/BEAM/Python+Tips#PythonTips-InstallingPythoninterpreters).
+- For large contributions, a signed [Individual Contributor License
+  Agreement](https://www.apache.org/licenses/icla.pdf) (ICLA) to the Apache
+  Software Foundation (ASF).
+
+### Share Your Intent
+1. Find or create an issue in the [Beam issue tracker (JIRA)](https://issues.apache.org/jira/projects/BEAM/issues).
+   Tracking your work in an issue will avoid duplicated or conflicting work, and provide
+   a place for notes. Later, your pull request will be linked to the issue as well.
+1. Assign the issue to yourself. To get the permission to do so, email
+   the [dev@beam.apache.org mailing list](https://beam.apache.org/community/contact-us)
+   to introduce yourself and to be added as a contributor in the Beam issue tracker including your
+   ASF Jira Username. For example [this welcome email](https://lists.apache.org/thread.html/e6018c2aaf7dc7895091434295e5b0fafe192b975e3e3761fcf0cda7@%3Cdev.beam.apache.org%3E).
+1. If your change is large or it is your first change, it is a good idea to
+   [discuss it on the dev@beam.apache.org mailing list](https://beam.apache.org/community/contact-us/).
+1. For large changes create a design doc
+   ([template](https://s.apache.org/beam-design-doc-template),
+   [examples](https://s.apache.org/beam-design-docs)) and email it to the [dev@beam.apache.org mailing list](https://beam.apache.org/community/contact-us/).
+
+### Setup Your Environment
+
```

Comment:
Add: 
Before you begin, check out the Wiki pages. There are many useful tips about [Git](https://cwiki.apache.org/confluence/display/BEAM/Git+Tips), [Go](https://cwiki.apache.org/confluence/display/BEAM/Go+Tips), [Gradle](https://cwiki.apache.org/confluence/display/BEAM/Gradle+Tips), [Java](https://cwiki.apache.org/confluence/display/BEAM/Java+Tips), [Python](https://cwiki.apache.org/confluence/display/BEAM/Python+Tips), etc.

Reply:
done